examples/dart/lib/services/account.dart

```diff
@@ -141,74 +140,74 @@ class Account extends Service {
+            'userId': userId,
+            'secret': secret,
+            'password-a': passwordA,
+            'password-b': passwordB,
+        };
 
-       return await this.client.call(HttpMethod.put, path: path, params: params);
+        return this.client.call(HttpMethod.put, path: path, params: params);
     }
      /// Get currently logged in user list of active sessions across different
      /// devices.
-    Future<Response> getSessions() async {
-       String path = '/account/sessions';
+    Future<Response> getSessions() {
+        final String path = '/account/sessions';
 
-       Map<String, dynamic> params = {
-       };
+        final Map<String, dynamic> params = {
+        };
 
-       return await this.client.call(HttpMethod.get, path: path, params: params);
+        return this.client.call(HttpMethod.get, path: path, params: params);
     }
      /// Allow the user to login into his account by providing a valid email and
      /// password combination. This route will create a new session for the user.
-    Future<Response> createSession({@required String email, @required String password}) async {
-       String path = '/account/sessions';
+    Future<Response> createSession({@required String email, @required String password}) {
+        final String path = '/account/sessions';
 
-       Map<String, dynamic> params = {
-         'email': email,
-         'password': password,
-       };
+        final Map<String, dynamic> params = {
+            'email': email,
+            'password': password,
+        };
 
-       return await this.client.call(HttpMethod.post, path: path, params: params);
+        return this.client.call(HttpMethod.post, path: path, params: params);
     }
      /// Delete all sessions from the user account and remove any sessions cookies
      /// from the end client.
-    Future<Response> deleteSessions() async {
-       String path = '/account/sessions';
+    Future<Response> deleteSessions() {
+        final String path = '/account/sessions';
 
-       Map<String, dynamic> params = {
-       };
+        final Map<String, dynamic> params = {
+        };
 
-       return await this.client.call(HttpMethod.delete, path: path, params: params);
+        return this.client.call(HttpMethod.delete, path: path, params: params);
     }
      /// Allow the user to login to his account using the OAuth2 provider of his
      /// choice. Each OAuth2 provider should be enabled from the Appwrite console
      /// first. Use the success and failure arguments to provide a redirect URL's
      /// back to your app when login is completed.
-    Future<Response> createOAuth2Session({@required String provider, @required String success, @required String failure}) async {
-       String path = '/account/sessions/oauth2/{provider}'.replaceAll(RegExp('{provider}'), provider);
+    Future<Response> createOAuth2Session({@required String provider, @required String success, @required String failure}) {
```

Comment:
On the web client (JS SDK) we use this method to just return the GET url required for login, like this:
https://github.com/appwrite/sdk-for-js/blob/master/src/sdk.js#L773

This is done because of the way OAuth protocol works.

Does this seems like a reasonable solution for flutter applications?

Reply:
This seems to be a little more complex than I thought. Gonna merge the PR and test this locally.

Reply:
There are some packages for authenticating via oauth2:
https://pub.dev/packages/oauth2